parallel-install/pkg/deployment/overrideinterceptors.go

```diff
@@ -38,7 +55,48 @@ func (i *DomainNameOverrideInterceptor) Intercept(value interface{}, key string)
 }
 
 func (i *DomainNameOverrideInterceptor) Undefined(overrides map[string]interface{}, key string) error {
-	return NewFallbackOverrideInterceptor(localKymaDevDomain).Undefined(overrides, key)
+	domain, err := i.discoverDomain()
+	if err != nil {
+		return err
+	}
+
+	return NewFallbackOverrideInterceptor(domain).Undefined(overrides, key)
+}
+
+func (i *DomainNameOverrideInterceptor) discoverDomain() (string, error) {
```

Comment:
This should probably also run even when the domain is not undefined right? Gardener domain needs to even override user provided custom domains right?